jupyterhub/singleuser.py

```diff
@@ -385,20 +391,30 @@ def _validate_static_custom_path(self, proposal):
             path = list(_exclude_home(path))
         return path
 
-    async def check_hub_version(self):
-        """Test a connection to my Hub
-
-        - exit if I can't connect at all
-        - check version and warn on sufficient mismatch
-        """
+    hub_http_client = Any()
```

Comment:
Down the road it may make sense in the contributor guide to highlight this pattern (traitlet and dynamic validation) or improve the traitlets doc for this.